### PR TITLE
[bug fix][api] sorting issue fix

### PR DESCRIPTION
**Pull request by ann-aot**
- Request table -> sorting was not properly working in some scenario
- History table -> submitted data sort as default (i.e ASC, updated to DESC)
image
image
image

### CODE REVIEW

**Review by henryoforeh-aot — APPROVED** [review]

File: sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts [hunk 1]
```diff
@@ -66,17 +66,29 @@ export class AssessmentAESTController extends BaseController {
...
-    );
+    // status and submitted date sorting
+    return requestedAssessments.sort((first, second) => {
+      if (
+        first.status === StudentAppealStatus.Pending &&
+        second.status === StudentAppealStatus.Pending &&
+        first.submittedDate > second.submittedDate
+      ) {
+        return -1;
+      } else if (
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts:77]: else if can be removed.

File: sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts [hunk 2]
```diff
@@ -66,17 +66,29 @@ export class AssessmentAESTController extends BaseController {
...
+        first.status === StudentAppealStatus.Pending &&
+        second.status === StudentAppealStatus.Declined
+      ) {
+        return -1;
+      } else if (
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts:82]: `else if` can be removed.

File: sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts [hunk 3]
```diff
@@ -66,17 +66,29 @@ export class AssessmentAESTController extends BaseController {
...
+        first.status === StudentAppealStatus.Declined &&
+        second.status === StudentAppealStatus.Declined &&
+        first.submittedDate > second.submittedDate
+      ) {
+        return -1;
+      } else {
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts:88]: `else` can be removed.

**Review by henryoforeh-aot — APPROVED** [review]

File: sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts [hunk 4]
```diff
@@ -66,17 +66,29 @@ export class AssessmentAESTController extends BaseController {
...
-    // sorting with status and returning
-    return requestedAssessments.sort((first) =>
-      first.status === StudentAppealStatus.Pending ||
-      first.status === ScholasticStandingStatus.Pending
-        ? -1
-        : 1,
-    );
+    // status and submitted date sorting
+    return requestedAssessments.sort((first, second) => {
+      if (
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts:71]: It seems that the logic can be simplified like:
```` typescript
if (first.status === second.status) {
   // check first.submittedDate > second.submittedDate
}
```

File: sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts [hunk 5]
```diff
@@ -66,17 +66,24 @@ export class AssessmentAESTController extends BaseController {
...
+        first.status === second.status &&
+        first.submittedDate > second.submittedDate
+      ) {
+        return -1;
+      }
+      if (
+        (first.status === StudentAppealStatus.Pending &&
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts:78]: This is could also be simplified as `first.status !== second.status`, right?

Comment (ann-aot) [reply sources/packages/api/src/route-controllers/assessment/assessment.aest.controller.ts:78]: no here, the first status should be pending and second declined, not the reverse

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for the clarifications, looks good 👍

**Review by guru-aot — APPROVED** [review]
LGTM

**Review by dheepak-aot — APPROVED** [review]
Looks good 👍

**Pull request merged**